instrumentation/play-ws/play-ws-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/playws/PlayWsClientHttpAttributesExtractor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.playws;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import play.shaded.ahc.org.asynchttpclient.Request;
+import play.shaded.ahc.org.asynchttpclient.Response;
+import play.shaded.ahc.org.asynchttpclient.uri.Uri;
+
+final class PlayWsClientHttpAttributesExtractor extends HttpAttributesExtractor<Request, Response> {
+
+  @Override
+  protected String method(Request request) {
+    return request.getMethod();
+  }
+
+  @Override
+  protected String url(Request request) {
+    return request.getUri().toUrl();
+  }
+
+  @Override
+  protected String target(Request request) {
+    Uri uri = request.getUri();
+    String query = uri.getQuery();
+    return query != null ? uri.getPath() + "?" + query : uri.getPath();
+  }
+
+  @Override
+  @Nullable
+  protected String host(Request request) {
+    String host = request.getHeaders().get("Host");
+    if (host != null) {
+      return host;
+    }
+    return request.getVirtualHost();
+  }
+
+  @Override
+  @Nullable
+  protected String scheme(Request request) {
+    return request.getUri().getScheme();
+  }
+
+  @Override
+  @Nullable
+  protected String userAgent(Request request) {
+    return null;
```

Comment:
It can be extracted from the headers, can't it?
```suggestion
    return request.getHeaders().get("User-Agent");
```

Reply:
Turns out it isn't that easy. PlayWs uses AHC and AHC only sets `User-Agent` on underlying netty request so getting access to this header requires more effort, see #3930. To me this seems like an unsolved problem with our current instrumentation strategy, if higher level library uses lower level library then higher level library instrumentation might not have access to all the details that lower level library instrumentation has.

Reply:
this might be a good use case for trying out the CLIENT span enrichment idea, of having nested CLIENT span update some attributes on the parent CLIENT span